CloseSpannerWithOpenResultSetTest is flaky

```
[ERROR] Tests run: 3, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 481.1 s <<< FAILURE! -- in com.google.cloud.spanner.CloseSpannerWithOpenResultSetTest
[ERROR] com.google.cloud.spanner.CloseSpannerWithOpenResultSetTest.testBatchClient_closedSpannerWithOpenResultSet_streamsAreCancelled -- Time elapsed: 480.3 s <<< ERROR!
com.google.cloud.spanner.SpannerException: RESOURCE_EXHAUSTED: io.grpc.StatusRuntimeException: RESOURCE_EXHAUSTED: Connection closed after GOAWAY. HTTP/2 error code: ENHANCE_YOUR_CALM (Bandwidth exhausted), debug data: too_many_pings - Statement: 'SELECT * FROM RANDOM'
```

--- a/google-cloud-spanner/src/test/java/com/google/cloud/spanner/CloseSpannerWithOpenResultSetTest.java
+++ b/google-cloud-spanner/src/test/java/com/google/cloud/spanner/CloseSpannerWithOpenResultSetTest.java
@@ -39,6 +39,8 @@ import java.util.concurrent.ThreadLocalRandom;
 import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
@@ -59,6 +61,16 @@ public class CloseSpannerWithOpenResultSetTest extends AbstractMockServerTest {
         .getService();
   }
 
+  @BeforeClass
+  public static void setWatchdogTimeout() {
+    System.setProperty("com.google.cloud.spanner.watchdogTimeoutSeconds", "1");
+  }
+
+  @AfterClass
+  public static void clearWatchdogTimeout() {
+    System.clearProperty("com.google.cloud.spanner.watchdogTimeoutSeconds");
+  }
+
   @After
   public void cleanup() {
     mockSpanner.unfreeze();
@@ -75,7 +87,13 @@ public class CloseSpannerWithOpenResultSetTest extends AbstractMockServerTest {
             client.batchReadOnlyTransaction(TimestampBound.strong());
         ResultSet resultSet = transaction.executeQuery(SELECT_RANDOM_STATEMENT)) {
       mockSpanner.freezeAfterReturningNumRows(1);
-      assertTrue(resultSet.next());
+      // This can sometimes fail, as the mock server may not always actually return the first row.
+      try {
+        assertTrue(resultSet.next());
+      } catch (SpannerException exception) {
+        assertEquals(ErrorCode.DEADLINE_EXCEEDED, exception.getErrorCode());
+        return;
+      }
       ((SpannerImpl) spanner).close(1, TimeUnit.MILLISECONDS);
       // This should return an error as the stream is cancelled.
       SpannerException exception = assertThrows(SpannerException.class, resultSet::next);
@@ -93,7 +111,13 @@ public class CloseSpannerWithOpenResultSetTest extends AbstractMockServerTest {
     try (ReadOnlyTransaction transaction = client.readOnlyTransaction(TimestampBound.strong());
         ResultSet resultSet = transaction.executeQuery(SELECT_RANDOM_STATEMENT)) {
       mockSpanner.freezeAfterReturningNumRows(1);
-      assertTrue(resultSet.next());
+      // This can sometimes fail, as the mock server may not always actually return the first row.
+      try {
+        assertTrue(resultSet.next());
+      } catch (SpannerException exception) {
+        assertEquals(ErrorCode.DEADLINE_EXCEEDED, exception.getErrorCode());
+        return;
+      }
       List<ExecuteSqlRequest> executeSqlRequests =
           mockSpanner.getRequestsOfType(ExecuteSqlRequest.class).stream()
               .filter(request -> request.getSql().equals(SELECT_RANDOM_STATEMENT.getSql()))

